paymentapi/src/main/java/fi/hel/verkkokauppa/payment/service/refund/PaytrailRefundPaymentService.java

```diff
@@ -0,0 +1,343 @@
+import java.util.Optional;
+
+@Service
+@Slf4j
+public class PaytrailRefundPaymentService {
+
+    @Autowired
+    private RefundPaymentRepository refundPaymentRepository;
+
+    @Autowired
+    private SendEventService sendEventService;
+
+    @Autowired
+    private SendNotificationService sendNotificationService;
+
+    @Autowired
+    private PaytrailPaymentContextBuilder paytrailPaymentContextBuilder;
+
+    @Autowired
+    private PaytrailPaymentClient paytrailPaymentClient;
+
+
+    public void isValidUserToCreateRefundPayment(String refundId, String userId) {
+        if (userId == null || userId.isEmpty()) {
+            log.warn("creating refund payment without user rejected, refundId: " + refundId);
+            throw new CommonApiException(
+                    HttpStatus.FORBIDDEN,
+                    new Error("rejected-creating-payment-for-order-without-user", "rejected creating refund payment for refund without user, refundId id [" + refundId + "]")
+            );
+        }
+    }
+
+    public void isValidRefundStatusToCreateRefundPayment(String refundId, String refundStatus) {
+        // check refund status, can only create refund payment for confirmed refunds
+        if (!"confirmed".equals(refundStatus)) {
+            log.warn("creating refund payment for unconfirmed refund rejected, refundId: " + refundId);
+            throw new CommonApiException(
+                    HttpStatus.FORBIDDEN,
+                    new Error("rejected-creating-refund-payment-for-unconfirmed-refund", "rejected creating refund payment for unconfirmed refund, refundId id [" + refundId + "]")
+            );
+        }
+    }
+
+    public void setPaymentStatus(String paymentId, String status) {
+        RefundPayment payment = getRefund(paymentId);
+        payment.setStatus(status);
+        refundPaymentRepository.save(payment);
+    }
+
+    public RefundPayment getRefundPaymentForOrder(String orderId) {
+        List<RefundPayment> payments = refundPaymentRepository.findByOrderId(orderId);
+
+        RefundPayment paidPayment = selectPaidRefundPayment(payments);
+        RefundPayment payablePayment = selectPayableRefundPayment(payments);
+
+        if (paidPayment != null) {
+            return paidPayment;
+        } else if (payablePayment != null) {
+            return payablePayment;
+        } else {
+            return null;
+        }
+    }
+
+    public List<RefundPayment> getPaymentsForOrder(String orderId, String namepace) {
+        return refundPaymentRepository.findByNamespaceAndOrderId(namepace, orderId);
+    }
+
+    public List<RefundPayment> getPaymentsWithNamespaceAndOrderIdAndStatus(String orderId, String namepace, String status) {
+        return refundPaymentRepository.findByNamespaceAndOrderIdAndStatus(namepace, orderId, status);
+    }
+
+
+    private RefundPayment selectPayableRefundPayment(List<RefundPayment> payments) {
+        RefundPayment payablePayment = null;
+
+        if (payments != null)
```

Comment:
Ihis if -statement is missing brackets.

Reply:
This is if -statement is actually not necessary, because **refundPaymentRepository.findByOrderId(orderId)** will never return null - spring data repository only returns empty array if no matches are found.

Reply:
Yeh we can assume that, but what happens if this function is called without spring data repository result which can be null and then can cause problems

Reply:
Very true. At the moment it was only used at those specific spots where the repository result is used as parameters. No need to change it if it is supposed to be a general purpose method.

Reply:
But yeh I tested this with automation test, and you are correct and I will remove this to make code easier to read